cpp/src/cupti.cpp

```diff
@@ -0,0 +1,424 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef RAPIDSMPF_HAVE_CUPTI
+
+#include <algorithm>
+#include <array>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+#include <unistd.h>
+
+#include <rapidsmpf/cupti.hpp>
+
+namespace rapidsmpf {
+
```

Comment:
Use an anonymous namespace instead of `static`.

```suggestion
namespace {
```